### PR TITLE
(core) make instance links customizable

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Allows configuration of instance links

(split out from https://github.com/spinnaker/deck/pull/2313)

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
@duftler @danielpeach @zanthrash PTAL

**Comment by danielpeach** [comment]
This does not work for GCE, but I think we've misconfigured the our instanceDetails controller. I can sort that out.

File: app/scripts/modules/google/instance/details/instance.details.controller.js [hunk 1]
```diff
@@ -128,11 +128,11 @@ module.exports = angular.module('spinnaker.instance.detail.gce.controller', [
           $scope.instance.region = region;
           $scope.instance.vpcId = vpcId;
           $scope.instance.loadBalancers = loadBalancers;
-          $scope.baseIpAddress = details.publicDnsName || details.privateIpAddress;
 
           $scope.instance.internalDnsName = $scope.instance.instanceId;
           $scope.instance.internalIpAddress = $scope.instance.networkInterfaces[0].networkIP;
           $scope.instance.externalIpAddress = $scope.instance.networkInterfaces[0].accessConfigs[0].natIP;
+          $scope.baseIpAddress = $scope.instance.internalIpAddress || $scope.instance.externalIpAddress || $scope.instance.internalDnsName;
```
Comment (anotherchrisberry) [inline app/scripts/modules/google/instance/details/instance.details.controller.js]: @danielpeach does this fix it on the Google side?

Comment (anotherchrisberry) [inline app/scripts/modules/google/instance/details/instance.details.controller.js]: Ha, just saw your other PR, will roll this back.

**Comment by jcogilvie** [comment]
Where's the doc for the variables we're able to use?  I was able to fumble around to find out that the link format can be something like:

`https://service.sumologic.com/ui/#section/search/@@instance%3D%22{{id}}%22` where `{{id}}` is interpolated.  Or `https://{{privateIpAddress}}:9000/gate/health`.  But what other values are available?

**Comment by anotherchrisberry** [comment]
No docs, unfortunately - we don't have clearly defined definitions on what the instance looks like across cloud providers. We'll eventually get some definition there as we continue converting everything to Typescript.

In the meantime (I know this is really far from ideal, but want to share in case you're trying to get these set up right now), you can figure out the shape of the instance by setting a breakpoint in the script:
1. With the app running, Open Chrome Dev Tools
2. Click on the "Sources" tab
3. Click on "app.js"
4. Search for the following text in app.js: `url.includes('{{')`
5. Set a breakpoint on that line
6. With Dev Tools still open, click on an instance (in an app with links configured) to pull up the details
7. It should hit that breakpoint, and you should get an inspector on the right of the Dev Tools.
8. In probably the second section of "Scope", you'll see "Closure (controller)". Expand that and look for `_this`. Expand `_this` and look for `instance`.
9. All the fields in `instance` are the ones that are resolvable for those links.

Which cloud provider are you deploying to? I can try to dig in this week and get some documentation together around what fields should be available...

**Comment by jcogilvie** [comment]
AWS is my provider.

**Comment by anotherchrisberry** [comment]
Added some documentation here: http://www.spinnaker.io/docs/customizing-instance-links

**Pull request merged**